tests/data/dataset_readers/dataset_reader_test.py

```diff
@@ -0,0 +1,29 @@
+# pylint: disable=no-self-use,abstract-method
+import pytest
+
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.params import Params
+from allennlp.data.dataset_readers import DatasetReader
+
+@DatasetReader.register('not-overridden')
+class NotOverridden(DatasetReader):
+    pass
+
+@DatasetReader.register('yes-overridden')
+class YesOverridden(DatasetReader):
+    @classmethod
+    def from_params(cls, params: Params) -> 'YesOverridden':
+        return YesOverridden()
+
+class TestDatasetReader:
+    def test_raises_without_override(self):
+        params = Params({'type': 'not-overridden'})
+
+        with pytest.raises(ConfigurationError) as exc:
```

Comment:
You could use the `match` or `message` argument to `pytest.raises` instead of keeping around the exception variable and checking it afterward.  Same on the model test.